contracts/governance/extensions/GovernorStorage.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
+
+import "../Governor.sol";
+
+/**
+ * @dev Extension of {Governor} that implements storage of proposal details. This modules also provides primitives for
+ * the enumerability of proposals.
+ *
+ * Usecases for this module include:
+ * - UIs that expore the proposal state without relying on event indexing.
+ * - For some L2 chains where storage is compared to calldata, using a the proposalId version of queue and execute
+ *   might me cheaper.
+ *
+ * _Available since v5.0._
+ */
+abstract contract GovernorStorage is Governor {
+    struct ProposalDetails {
+        address[] targets;
+        uint256[] values;
+        bytes[] calldatas;
+        bytes32 descriptionHash;
+    }
+
+    uint256[] private _proposalIds;
+    mapping(uint256 => ProposalDetails) private _proposalDetails;
+
+    /**
+     * @dev See {IGovernor-propose}.
+     */
+    function propose(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description
+    ) public virtual override returns (uint256) {
+        uint256 proposalId = super.propose(targets, values, calldatas, description);
+        // store
+        _proposalIds.push(proposalId);
+        _proposalDetails[proposalId] = ProposalDetails({
+            targets: targets,
+            values: values,
+            calldatas: calldatas,
+            descriptionHash: keccak256(bytes(description))
+        });
+        return proposalId;
+    }
+
+    /**
+     * @dev ProposalId version of {IGovernorTimelock-queue}.
+     */
+    function queue(uint256 proposalId) public virtual {
+        ProposalDetails storage proposalDetails = _proposalDetails[proposalId];
+        queue(
```

Comment:
Is this equivalent to `super.queue(...)`? Asking because I genuinely don't know if it makes any difference in the inheritance order

Reply:
I think its different. Lets say you inherit: `MyGovernor is Governor, GovernorStorage, GovernorTimelockControl`

- calling `queue(...)` will get the public function, which is the result of the `Governor → GovernorTimelockControl` override. So we get the function we want.
- calling `super.queue(...)` will get the version of the parent, which here is `Governor`. It would skip the one in `GovernorTimelockControl` ... and revert.

If you inherit `GovernorTimelockControl` before `GovernorStorage` both versions will work, but we can't count on that. I think this is the way to do it.

Nice question though!

Reply:
It is different, here it's important to not use super.